django/forms/widgets.py

```diff
@@ -1014,7 +1014,8 @@ def format_value(self, value):
                     year, month, day = d.year, d.month, d.day
             match = self.date_re.match(value)
             if match:
-                year, month, day = [int(val) for val in match.groups()]
+                # In case of pseudo-ISO dates, e.g. 2017-0-23, return empty string to match empty option value.
```

Comment:
@ziima I know you didn't change this logic but, is this right? 

It looks to me as if we're parsing `value` into a `datetime` and then parsing it **again** using `date_re`. (Or did I read that wrong?) 

Shouldn't we be doing one or the other, and not both?

Reply:
I was startled as well, the string is parsed twice. Since this wasn't part of the code which would be subject of the ticket, I left it unchanged.